File: SwiftCallObject-C/SwiftCallObject-C/Person.m
//
//  Person.m
//  SwiftCallObject-C
//
//  Created by baijiawei on 15/5/14.
//  Copyright (c) 2015年 GarveyCalvin. All rights reserved.
//

#import "Person.h"
#import "SwiftCallObject-C-Bridging-Header.h"

@implementation Person

- (void)print
{
    NSLog(@"name is %@", self.name);
}

+ (id)personWithName:(NSString *)name
{
    Person *person = [[Person alloc] init];
    person.name = name;
    return person;
}

@end

File: Object_CCallSwift/Object_CCallSwift/Person.swift
//
//  Person.swift
//  Object_CCallSwift
//
//  Created by baijiawei on 15/5/17.
//  Copyright (c) 2015年 GarveyCalvin. All rights reserved.
//

import Cocoa

@objc class Person: NSObject {
    var name: String!
    
    func print() {
        Swift.print("person name is \(name)")
    }
}
